controllers/profiles/getMyProfile.js

```diff
@@ -1,6 +1,43 @@
 const {User, Locations, sequelize} = require('../../databases/models');
 
 const {checkMoreOrLess} = require('../../helpers/checkMoreOrLess');
+const UsersFunction = require('../../databases/functions/users');
+const {getDb} = require('../../databases/config/mongodb_conn');
+const {DB_COLLECTION_POST_SCORE} = require('../../services/score/constant');
+
+const countAuthorPost = async (authorId, anonAuthorId) => {
+  const userIds = [authorId, anonAuthorId];
+  const db = await getDb();
+  const postsCollection = db.collection(DB_COLLECTION_POST_SCORE);
+  const count = await postsCollection.countDocuments({author_id: {$in: userIds}});
+
+  return count;
+};
+
+const checkIsKarmaUnlocked = async (userId) => {
+  // get anonymous_user_id
+  const anonymousUser = await UsersFunction.findAnonymousUserId(User, userId, {raw: true});
+  // check whether the user has ever made a post in DB
+  const hasPostQuery = `SELECT COUNT(*) as count_post from posts WHERE author_user_id in (:user_id) limit 1`;
+  const hasPost = await sequelize.query(hasPostQuery, {
+    replacements: {user_id: [userId, anonymousUser?.user_id]}
+  });
+  const hasPostResult = hasPost?.[0]?.[0]?.count_post;
+  let countPostInMongo = await countAuthorPost(userId, anonymousUser?.user_id);
+
+  return parseInt(hasPostResult) + parseInt(countPostInMongo) > 0;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `checkIsKarmaUnlocked` function is making two separate database queries to check if a user has made any posts. This could be optimized by combining these into a single query, which would reduce the load on your databases and potentially improve performance.
```diff
-  const hasPostQuery = `SELECT COUNT(*) as count_post from posts WHERE author_user_id in (:user_id) limit 1`;
-  const hasPost = await sequelize.query(hasPostQuery, {
-    replacements: {user_id: [userId, anonymousUser?.user_id]}
-  });
-  const hasPostResult = hasPost?.[0]?.[0]?.count_post;
-  let countPostInMongo = await countAuthorPost(userId, anonymousUser?.user_id);
-
-  return parseInt(hasPostResult) + parseInt(countPostInMongo) > 0;
+  const userIds = [userId, anonymousUser?.user_id];
+  const db = await getDb();
+  const postsCollection = db.collection(DB_COLLECTION_POST_SCORE);
+  const count = await postsCollection.countDocuments({author_id: {$in: userIds}});
+
+  return count > 0;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->